agent/consul/leader_metrics.go

```diff
@@ -2,25 +2,40 @@ package consul
 
 import (
 	"context"
+	"crypto/x509"
+	"errors"
 	"fmt"
 	"time"
 
+	"github.com/hashicorp/consul/agent/connect/ca"
+
+	"github.com/hashicorp/consul/agent/connect"
+
 	"github.com/armon/go-metrics"
 	"github.com/armon/go-metrics/prometheus"
-	"github.com/hashicorp/go-hclog"
-
 	"github.com/hashicorp/consul/logging"
+	"github.com/hashicorp/go-hclog"
 )
 
+var metricsKeyMeshRootCAExpiry = []string{"mesh", "active-root-ca", "expiry"}
+var metricsKeyMeshPrimaryCAExpiry = []string{"mesh", "active-primary-dc-ca", "expiry"}
+var metricsKeyMeshSecondaryCAExpiry = []string{"mesh", "active-secondary-dc-ca", "expiry"}
+
 var CertExpirationGauges = []prometheus.GaugeDefinition{
 	{
 		Name: metricsKeyMeshRootCAExpiry,
-		Help: "Seconds until the service mesh root certificate expires.",
+		Help: "Seconds until the service mesh root certificate expires. Updated every hour",
+	},
+	{
+		Name: metricsKeyMeshPrimaryCAExpiry,
+		Help: "Seconds until the service mesh primary DC certificate expires. Updated every hour",
+	},
+	{
+		Name: metricsKeyMeshSecondaryCAExpiry,
+		Help: "Seconds until the service mesh secondary DC certificate expires. Updated every hour",
```

Comment:
I think the only question I have left is if we should report two separate metrics, or a single one called `active-signing-ca`. I think either could work.The datacenter label should work to differentiate the values from different DCs.

I'm starting to like the idea of using a single metric. I expect it will be a little easier to set up monitoring if both primary and secondary use the same metric name. It might be worth it to see what others think about these two options.

Reply:
Yes I think you're right a single metric that would return the right expiry will be easier to integrate and expose the user less to the implementation details
I will make the change